modern/src/DevicesList.js

```diff
@@ -26,8 +28,72 @@ const useStyles = makeStyles(() => ({
     height: '25px',
     filter: 'brightness(0) invert(1)',
   },
+  batteryText: {
+    fontSize: '0.75rem',
+    fontWeight: 'normal',
+    lineHeight: '0.875rem'
+  },
+  green: {
+    color: theme.palette.common.green,
+  },
+  red: {
+    color: theme.palette.common.red,
+  },
+  gray: {
+    color: theme.palette.common.gray
+  }
 }));
 
+const OnlineStatus =({ status }) => {
+  
+  const classes = useStyles();
+
+  switch (status) {
+    case 'online':
+      return <span className={classes.green}>{status}</span>
+    case 'offline':
+      return <span className={classes.red}>{status}</span>
+    case 'unknown':
+    default:
+      return <span className={classes.gray}>{status}</span>
+  }
+}
+
+const DeviceStatus = ({ deviceId }) => {
+  const classes = useStyles();
+  let batteryClass = '';
+  
+  const position = useSelector(state => state.positions.items[deviceId]);
+  if (!position) {
+    return null;
+  }
+  const batteryLevel = position.attributes.hasOwnProperty('batteryLevel') ? position.attributes.batteryLevel : 'undefined';
+  const ignitionStatus = position.attributes.hasOwnProperty('ignition') ? position.attributes.ignition : 'undefined';
+
+  if (batteryLevel >= 70) {
+    batteryClass = classes.green;
+  } else if (batteryLevel > 30) {
+    batteryClass = classes.gray;
+  } else {
+    batteryClass = classes.red;
+  }
+  return (
+    <Grid container direction="row" alignItems="center" alignContent="center" spacing={1}>
+      {ignitionStatus !== 'undefined' && <Grid item>
```

Comment:
Instead of this can we just use `position.attributes.hasOwnProperty('ignition')`? That way we don't need extra variabled.

Reply:
Done.